File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }

        Set<Integer> uniqueValues = new HashSet();

        for (int num : nums) {
            uniqueValues.add(num);
        }

        int longestSequence = 1;
        int currentSequence = 1;

        for (int num : nums) {
            if (!uniqueValues.contains(num-1)) {
                int next = num+1;
                while (uniqueValues.contains(next)) {
                    next++;
                    currentSequence++;
                }
                if (currentSequence > longestSequence) longestSequence = currentSequence;
                currentSequence = 1;
            }
        }

        return longestSequence;
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        k, i = 0, 0

        while i < len(nums):
            if i == 0 or nums[i] != nums[i-1]:
                nums[k] = nums[i]
                k += 1
            i += 1
        
        return k